### PR TITLE
Dev/agilent34450a

### PR DESCRIPTION
**Pull request by theMashUp**
Adds instrument Agilent34450A Digital Multimeter.

### CODE REVIEW

File: pymeasure/instruments/agilent/agilent34450A.py [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+
+        # Eliminate empty string elements
+        list_without_empty_elements = list(filter(lambda v: v != '', list_of_elements))
+
+        # Convert numbers from str to float, where applicable
+        for i, v in enumerate(list_without_empty_elements):
+            try:
+                list_without_empty_elements[i] = float(v)
+            except ValueError as e:
+                print(e)
```
Comment (bilderbuchi) [inline pymeasure/instruments/agilent/agilent34450A.py:565]: that should probably use the `log` you define above, instead of `print`

**Review by bilderbuchi — CHANGES_REQUESTED** [review]
I gave this a cursory look, noticed nothing critical (only one minor thing). Did you test (all) the functionality with a real device?

File: pymeasure/instruments/agilent/agilent34450A.py (hunk 1 above)
Comment (theMashUp) [reply pymeasure/instruments/agilent/agilent34450A.py]: Agreed, modified in new commit e0e1a7a.

**Comment by theMashUp** [comment]
All functions were tested with the instrument, I took out the tests with commit 5539936 to not mess up with the normal testing routine when the device is not connected.

**Comment by bilderbuchi** [comment]
Nice work on the large number of tests! I'm hesitant to just throw this away - alternatively, could you mark the tests as skipped instead? Using `pytest.skip('Only work with connected hardware', allow_module_level=True)` at the top of the file should skip all the tests, but would keep them around.
In case we implement a better test suite even for non-connected hardware (see #249), we could then enable/adapt these tests. 
(You'll need to fix the merge conflict in the authors file first so that CI can run correctly)

**Comment by theMashUp** [comment]
Re-added the tests with the skip. I corrected the AUTHORS.txt conflict, but it still shows up somehow, I am not sure what is up with that.

File: tests/instruments/agilent/test_agilent34450A.py [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+    # TESTS #
+    #########
+
+    def test_dmm_initialization_bad(self):
+        bad_resource = "USB0::10893::45848::MY12345678::0::INSTR"
+        with pytest.raises(VisaIOError):
+            dmm = Agilent34450A(bad_resource)
+
+    def test_good_address(self):
+        dmm = Agilent34450A(self.resource)
```
Comment (bilderbuchi) [inline tests/instruments/agilent/test_agilent34450A.py:89]: can you add an assert here, e.g. query a status or `.id` or something to confirm communicaiton with the device actually works, or is this considered successful if it doesn't raise an exception? If the latter, maybe add a comment about this.

File: tests/instruments/agilent/test_agilent34450A.py [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+
+    def test_reset(self):
+        dmm = Agilent34450A(self.resource)
+        dmm.reset()
```
Comment (bilderbuchi) [inline tests/instruments/agilent/test_agilent34450A.py:93]: Is there some query that you can use to assert the successful reset here? A boot cycle counter? Prior to the reset setting a quantity that gets reset by the device `reset()`, then checking if that happened?

**Comment by bilderbuchi** [comment]
Don't worry, I'll sofrt out the authors.txt for you.

**Comment by bilderbuchi** [comment]
Oh, you used f-strings, which are incompatible with python 3.5. I think it is time to remove support for EOL'd python3.5, see #338, so this should be fine for now.

There is a warning in the test log:
```
pymeasure/instruments/agilent/agilent34450A.py:555

  /home/travis/build/ralph-group/pymeasure/pymeasure/instruments/agilent/agilent34450A.py:555: DeprecationWarning: invalid escape sequence \s

    list_of_elements = re.split('["\s,]', one_long_string)
```

Pytest does not show the skipped tests, but that is apparently a [known issue](https://github.com/pytest-dev/pytest/issues/5551).

**Comment by bilderbuchi** [comment]
I removed Python 3.5 support from master, so with your next modifications, do a merge/rebase from pymeasure master and the CI should start to work again. :-)

File: tests/instruments/agilent/test_agilent34450A.py [hunk 4]
```diff
@@ -76,28 +70,35 @@ def make_reseted_dmm(self):
...
+        return self.DMM
+
     #########
     # TESTS #
     #########
 
     def test_dmm_initialization_bad(self):
         bad_resource = "USB0::10893::45848::MY12345678::0::INSTR"
-        with pytest.raises(VisaIOError):
+        # The pure python VISA library (pyvisa-py) raises a ValueError while the PyVISA library raises a VisaIOError.
```
Comment (bilderbuchi) [inline tests/instruments/agilent/test_agilent34450A.py:89]: huh, weird, that inconsistency might be worth to report a bug against pyvisa-py...

**Comment by bilderbuchi** [comment]
Thank you for your effort! Merging this now.

**Pull request merged**